### ISSUE TITLE
Define product requirements for PDF generation of submitted forms

### ISSUE THREAD

**Issue by neel-a-mehta** [seattle-uat/civiform#546]
## Situation
The normal way to apply for government programs is through a paper form or a printable PDF form.

We want to make it so that, every time an applicant hits Submit, their answers get automatically inserted into such a PDF, which is then sent off to the program admins. The program admin would specify which questions go to which sections of the PDF, but how to actually convert user data into an outputted PDF is a hard question.

## What we need
We need someone to write up a product requirements doc that explains how this process of `answers => finished PDF` would work. Include answers to questions including: 

- What technology would we use to insert answers into a PDF? 
- What happens if the user's answers are too long to fit into the PDF's form fields? We would probably make an overflow page and attach it at the end of the PDF, but what would that look like and how would the main PDF point to the right overflow page?
- What happens when we have repeated fields? How do those get automatically populated into the right PDF fields? What happens if we have more repeated fields than the PDF form can normally support?
- How would uploaded images be inserted into the PDF? On an addendum page? Or shrunk and inserted into a particular box?
- How do we leverage the fact that Adobe Acrobat [can specify certain form fields in a PDF as "fillable"](https://acrobat.adobe.com/us/en/acrobat/how-to/create-fillable-pdf-forms-creator.html) to make our lives easier?
- For checkbox / radio button questions, how do we use the user's answer to figure out which box on the PDF form to check?
- What do we put in the signature or initials fields found at the bottom of so many PDFs?

Basically, we'd like to get a list of answers to many questions we'll run into when trying to develop a system for converting user answers into filled PDFs. The more answers you can give, the better!

## Resources
First, look at a couple government PDF forms like [this](https://www.seattle.gov/Documents/Departments/ParksAndRecreation/Programs/ENGLISH%20General%20Scholarship%20Application%202021-2022_B.pdf) and [this](https://www.dshs.wa.gov/sites/default/files/FSA/forms/pdf/14-001.pdf) to see how government PDFs are usually laid out.

We have a lot of [issues](https://github.com/seattle-uat/civiform/labels/pdf) around PDF generation already, so check those out.

**Comment by neel-a-mehta** [comment]
Anything to add, @gcapiel @ndmckinley ?
